src/main/java/crazypants/enderio/machine/vacuum/TileVacuumChest.java

```diff
@@ -191,9 +226,86 @@ public boolean isItemValidForSlot(int var1, ItemStack var2) {
     return true;
   }
 
+  public boolean isItemValidForFilter(ItemStack itemstack) {
+    return itemstack != null && itemstack.getItem() == EnderIO.itemBasicFilterUpgrade && itemstack.getItemDamage() == 0;
+  }
+
+  public int getRange() {
+    return range;
+  }
+
+  private int limitRange(int range) {
+    return Math.max(1, Math.min(Config.vacuumChestRange, range));
```

Comment:
enable / disable is handled via redstone control in the 2nd patch. Setting range to 0 would only provide a 2nd way to disable it - and also require a special check in doHover or it would waste CPU.